src/source/codegen.js

```diff
@@ -13,5 +13,7 @@ startJSONStream({
     //
     // This one takes an AST from the Rust side as an input and applies
     // a codegen as a transform, returning generated JavaScript back.
-    transform: ast => codegen(ast)
+    transform: ast => codegen(ast),
+
+    inputIsEscapedWTF8: true,
```

Comment:
Is this really necessary or would it have the same effect to do `unescapeWTF8(codegen(ast))` in the `transform`?

Reply:
Actually, now I wonder the same for input to parse functions: could we get away with escaping source code itself via WTF-8, and then pass it to the parser, instead of escaping the resulting JSON?

Reply:
I added it there because we already have escapeWTF8 there.

Reply:
Fair enough, but in that case we always unescape, that's why it's in the generic interface. Although, not that I think of it, probably we shouldn't be escaping result of the codegen?

Reply:
can you rephrase?
I don't understand your comment.

Reply:
Yeah sorry, my train of thought just keeps going on...

What I'm thinking about now (past my last comment) is... but can we even realistically unescape WTF-8 and pass it back to Rust?

I mean, Rust expects a valid UTF-8 string from the codegen to write back to the file so it can't really represent surrogates and such anyway and won't be able to serialise these to JS...

Reply:
Oh, I see, you're transforming surrogates back to unicode escapes. I guess that's fine assuming they're inside of string literals.

Reply:
Then back to `escapeWTF8` - I'm just thinking that it's actually wrong to have it in the generic interface as well and needs to be moved to separate module used only by `parse_str` / `parse_file`.